.infrastructure/helm-chart/todoapp/templates/hpa.yml

```diff
@@ -0,0 +1,25 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Chart.Name }}-hpa
+  namespace: {{ .Values.todoapp.namespace }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: todoapp
+  minReplicas: {{ .Values.todoapp.hpa.minReplicas }}
+  maxReplicas: {{ .Values.todoapp.hpa.maxReplicas }}
```

Comment:
Ensure that the `todoapp.hpa.maxReplicas` value is defined in your `values.yaml` file to avoid issues with the HPA configuration.